src/lightning_app/core/work.py

```diff
@@ -207,6 +217,19 @@ def name(self):
         """Returns the name of the LightningWork."""
         return self._name
 
+    @property
+    def display_name(self):
+        """Returns the display name of the LightningWork in the cloud.
+
+        The display name needs to set before the run method of the work is called.
+        """
+        return self._display_name
+
+    @display_name.setter
```

Comment:
Continuing to what Adrian mentioned above, we shouldn't allow setting the name if it is after the run method started

Reply:
Answered above.